src/sidebar/test/cross-origin-rpc-test.js

```diff
@@ -78,9 +79,7 @@ describe('sidebar/cross-origin-rpc', () => {
         source: frame,
       });
 
-      assert.isTrue(
-        fakeStore.changeFocusModeUser.calledWithExactly('one', 'two')
-      );
```

Comment:
Introducing the wrapping of the two store methods with a single RPC method makes this difficult to test directly: that the registered method gets called with all of the parameters in `params`. That is, the new registered method looks like this:

```
changeFocusModeUser: userInfo => {
      store.changeFocusModeUser(userInfo);
      store.filterGroups(userInfo?.groups);
    },
```

The wrapping `changeFocusModeUser` _is_ called with all of the spread parameters (in this test's example, `['one', 'two']`), but only the first parameter is used and passed through to the store methods.

I spent a _while_ trying to make this more testable. I tried adding a test seam to the `startServer` method that would allow me to pass my own `registerMethods` function, which worked in tests but blew up in real life because of the need to register any arguments to an injected function with the `Injector`. I considered externalizing the `registeredMethods` function and mocking it in tests, but that seemed a little perverse and over-designed.

In the end I decided that testing the abstract notion of a registered method being called with "all of the parameters" — when in fact none of our code actually makes use of this — was outweighed by the PITA factor of testing it.

Raising this in case you think either A) no, we really need to have this test or B) you think this testing problem indicates a code smell.

Reply:
With the current architecture of the code, the concept of a "registered RPC method" is an implementation detail and so these tests were not really appropriate in the first place. The inputs into this module are RPC messages and the outputs are calls to store methods, or to just ignore the message if unknown. That's all we should really be testing.

Separating the wire protocol logic and application-level integration could be a sensible separation in future. This might have some benefits such as enabling the client and lms to use the same implementation of a JSON-RPC-over-postMessage server. The implementation in the LMS app does already have this separation.